{
  "nbformat": 4,
  "nbformat_minor": 5,
  "metadata": {
    "kernelspec": {
      "display_name": "R",
      "language": "R",
      "name": "ir"
    },
    "language_info": {
      "codemirror_mode": "r",
      "file_extension": ".r",
      "mimetype": "text/x-r-source",
      "name": "R",
      "pygments_lexer": "r",
      "version": "3.6.3"
    },
    "colab": {
      "name": "03_[R]_calculate_pie_charts.ipynb",
      "provenance": [],
      "collapsed_sections": []
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "scrolled": true,
        "id": "2645e505"
      },
      "source": [
        "if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
        "  install.packages(\"BiocManager\")\n",
        "BiocManager::install(\"ChIPpeakAnno\")\n",
        "BiocManager::install(\"ChIPseeker\")\n",
        "BiocManager::install(\"clusterProfiler\")\n",
        "library(GenomicFeatures)\n",
        "library(ChIPpeakAnno)\n",
        "library(ChIPseeker)\n",
        "library(clusterProfiler)"
      ],
      "id": "2645e505",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "3305c143"
      },
      "source": [
        "download.file(\"http://anonymous@ftp.ensembl.org/pub/grch37/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz\", destfile=\"Homo_sapiens.GRCh37.87.gtf.gz\", method='wget')"
      ],
      "id": "3305c143",
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FJzo_zAHSr6i",
        "outputId": "28c8b875-ae5a-49da-a07a-bd3b5e72f3c4"
      },
      "source": [
        "txdb_all_annotation <- makeTxDbFromGFF('Homo_sapiens.GRCh37.87.gtf.gz')"
      ],
      "id": "FJzo_zAHSr6i",
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Import genomic features from the file as a GRanges object ... \n",
            "OK\n",
            "\n",
            "Prepare the 'metadata' data frame ... \n",
            "OK\n",
            "\n",
            "Make the TxDb object ... \n",
            "Warning message in .get_cds_IDX(mcols0$type, mcols0$phase):\n",
            "\"The \"phase\" metadata column contains non-NA values for features of type\n",
            "  stop_codon. This information was ignored.\"\n",
            "OK\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "t-43OwgVSac0",
        "outputId": "59c8b46e-c585-416d-c05a-19708b25a90e"
      },
      "source": [
        "chip_list <- list(\n",
        "  'AGO2 HepG2.bed',\n",
        "  'AGO1 K562.bed',\n",
        "  'AGO1  HepG2.bed'\n",
        ")\n",
        "\n",
        "for (chip in chip_list) {\n",
        "  print('--------------------------------------------')\n",
        "  print(chip)\n",
        "  print('--------------------------------------------')\n",
        "  peakAnno <- annotatePeak(chip, tssRegion=c(-3000, 3000), TxDb=txdb_all_annotation)\n",
        "  print(peakAnno)\n",
        "}"
      ],
      "id": "t-43OwgVSac0",
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] \"--------------------------------------------\"\n",
            "[1] \"AGO2 HepG2.bed\"\n",
            "[1] \"--------------------------------------------\"\n",
            ">> loading peak file...\t\t\t\t 2022-05-13 05:35:53 PM \n",
            ">> preparing features information...\t\t 2022-05-13 05:35:53 PM \n",
            ">> identifying nearest features...\t\t 2022-05-13 05:35:53 PM \n",
            ">> calculating distance from peak to TSS...\t 2022-05-13 05:35:54 PM \n",
            ">> assigning genomic annotation...\t\t 2022-05-13 05:35:54 PM \n",
            ">> assigning chromosome lengths\t\t\t 2022-05-13 05:35:59 PM \n",
            ">> done...\t\t\t\t\t 2022-05-13 05:35:59 PM \n",
            "Annotated peaks generated by ChIPseeker\n",
            "46107/46107  peaks were annotated\n",
            "Genomic Annotation Summary:\n",
            "              Feature  Frequency\n",
            "9    Promoter (<=1kb) 54.6771640\n",
            "10   Promoter (1-2kb)  9.1981695\n",
            "11   Promoter (2-3kb)  6.5803457\n",
            "4              5' UTR  0.4185915\n",
            "3              3' UTR  4.8669399\n",
            "1            1st Exon  0.1518208\n",
            "7          Other Exon  4.2856833\n",
            "2          1st Intron  3.9907172\n",
            "8        Other Intron  9.6731516\n",
            "6  Downstream (<=300)  0.2233934\n",
            "5   Distal Intergenic  5.9340230\n",
            "[1] \"--------------------------------------------\"\n",
            "[1] \"AGO1 K562.bed\"\n",
            "[1] \"--------------------------------------------\"\n",
            ">> loading peak file...\t\t\t\t 2022-05-13 05:35:59 PM \n",
            ">> preparing features information...\t\t 2022-05-13 05:35:59 PM \n",
            ">> identifying nearest features...\t\t 2022-05-13 05:35:59 PM \n",
            ">> calculating distance from peak to TSS...\t 2022-05-13 05:35:59 PM \n",
            ">> assigning genomic annotation...\t\t 2022-05-13 05:35:59 PM \n",
            ">> assigning chromosome lengths\t\t\t 2022-05-13 05:36:04 PM \n",
            ">> done...\t\t\t\t\t 2022-05-13 05:36:04 PM \n",
            "Annotated peaks generated by ChIPseeker\n",
            "11428/11428  peaks were annotated\n",
            "Genomic Annotation Summary:\n",
            "              Feature  Frequency\n",
            "9    Promoter (<=1kb) 65.1120056\n",
            "10   Promoter (1-2kb)  8.3916696\n",
            "11   Promoter (2-3kb)  5.6440322\n",
            "4              5' UTR  0.2012601\n",
            "3              3' UTR  3.6314316\n",
            "1            1st Exon  0.1750088\n",
            "7          Other Exon  2.5901295\n",
            "2          1st Intron  1.8988449\n",
            "8        Other Intron  3.7014351\n",
            "6  Downstream (<=300)  0.2800140\n",
            "5   Distal Intergenic  8.3741687\n",
            "[1] \"--------------------------------------------\"\n",
            "[1] \"AGO1  HepG2.bed\"\n",
            "[1] \"--------------------------------------------\"\n",
            ">> loading peak file...\t\t\t\t 2022-05-13 05:36:04 PM \n",
            ">> preparing features information...\t\t 2022-05-13 05:36:04 PM \n",
            ">> identifying nearest features...\t\t 2022-05-13 05:36:04 PM \n",
            ">> calculating distance from peak to TSS...\t 2022-05-13 05:36:05 PM \n",
            ">> assigning genomic annotation...\t\t 2022-05-13 05:36:05 PM \n",
            ">> assigning chromosome lengths\t\t\t 2022-05-13 05:36:09 PM \n",
            ">> done...\t\t\t\t\t 2022-05-13 05:36:09 PM \n",
            "Annotated peaks generated by ChIPseeker\n",
            "13996/13996  peaks were annotated\n",
            "Genomic Annotation Summary:\n",
            "              Feature   Frequency\n",
            "9    Promoter (<=1kb) 70.17004859\n",
            "10   Promoter (1-2kb)  7.04486996\n",
            "11   Promoter (2-3kb)  4.55844527\n",
            "4              5' UTR  0.17147757\n",
            "3              3' UTR  2.57930837\n",
            "1            1st Exon  0.07859388\n",
            "7          Other Exon  1.87910832\n",
            "2          1st Intron  1.86481852\n",
            "8        Other Intron  3.29379823\n",
            "6  Downstream (<=300)  0.20720206\n",
            "5   Distal Intergenic  8.15232924\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "3lXglTADc1mr"
      },
      "id": "3lXglTADc1mr",
      "execution_count": null,
      "outputs": []
    }
  ]
}
